test/test_proxy_lib.cpp

```diff
@@ -15,21 +15,66 @@
 
 #include "base.hpp"
 #include "test_helpers.h"
+#include "utils_common.h"
 
 using umf_test::test;
 
-TEST_F(test, proxyLibBasic) {
+#define SIZE_64 64
+#define ALIGN_1024 1024
 
-    ::free(::malloc(64));
+TEST_F(test, proxyLib_basic) {
+
+    ::free(::malloc(SIZE_64));
 
     // a check to verify we are running the proxy library
     void *ptr = (void *)0x01;
+
 #ifdef _WIN32
     size_t size = _msize(ptr);
 #elif __APPLE__
     size_t size = ::malloc_size(ptr);
 #else
     size_t size = ::malloc_usable_size(ptr);
 #endif
+
     ASSERT_EQ(size, 0xDEADBEEF);
 }
+
+TEST_F(test, proxyLib_realloc_size0) {
+    // realloc(ptr, 0) == free (ptr)
+    // realloc(ptr, 0) returns NULL
+    ASSERT_EQ(::realloc(::malloc(SIZE_64), 0), nullptr);
+}
+
+TEST_F(test, proxyLib_malloc_usable_size) {
+
+    void *ptr = ::malloc(SIZE_64);
+
+#ifdef _WIN32
+    size_t size = _msize(ptr);
```

Comment:
not sure if we should care `'ptr' could be '0'`

Reply:
Fixed.
Done.